File: VB/WindowsApplication3/MVVM/Views/MainView.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Utils.MVVM.Services
Imports DevExpress.XtraEditors

Namespace MVVMSample
    Partial Public Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            If Not mvvmContext.IsDesignMode Then
                InitializeBindings()
            End If
        End Sub

        Private Sub InitializeBindings()
            documentTypeComboBoxEdit.Properties.Items.AddEnum(Of DocumentType)()
            mvvmContext.ViewModelType = GetType(DocumentsViewModel)
            mvvmContext.RegisterDefaultService(DocumentManagerService.Create(navigationFrame))
            Dim fluentAPI = mvvmContext.OfType(Of DocumentsViewModel)()
            fluentAPI.SetBinding(documentTypeComboBoxEdit, Function(c) c.EditValue, Function(x) x.DocumentType)
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication3/MVVM/Views/MainView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Utils.MVVM.Services;
using DevExpress.XtraEditors;

namespace MVVMSample {
    public partial class Main : XtraForm {
        public Main() {
            InitializeComponent();
            if (!mvvmContext.IsDesignMode)
                InitializeBindings();
        }

        void InitializeBindings() {
            documentTypeComboBoxEdit.Properties.Items.AddEnum<DocumentType>();
            mvvmContext.ViewModelType = typeof(DocumentsViewModel);
            mvvmContext.RegisterDefaultService(DocumentManagerService.Create(navigationFrame));
            var fluentAPI = mvvmContext.OfType<DocumentsViewModel>();
            fluentAPI.SetBinding(documentTypeComboBoxEdit, c => c.EditValue, x => x.DocumentType);
        }
    }
}

